From 52685499dfbd83971a95adf06a3baa24bfadbbbf Mon Sep 17 00:00:00 2001
From: Suresh Gupta <suresh.gupta@xilinx.com>
Date: Fri, 12 Oct 2018 15:21:42 +0530
Subject: [PATCH 0474/1851] v4l: xilinx-multi-scaler: Read tap from the
 device-tree

commit cd8bea636392bbbc099546e60db98820d0dd379f from
https://github.com/Xilinx/linux-xlnx.git

Previously the tap value is hardcoded to 6 in code.
This patch read the tap value from DT.

Signed-off-by: Suresh Gupta <suresh.gupta@xilinx.com>
Reviewed-by: Saurabh Sengar <saurabh.singh@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-multi-scaler.c | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-multi-scaler.c b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
index 774559b8cb0d..21e61b5782fc 100644
--- a/drivers/media/platform/xilinx/xilinx-multi-scaler.c
+++ b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
@@ -550,8 +550,6 @@ static void xm2msc_set_chan_com_params(struct xm2msc_chan_ctx *chan_ctx)
 	u32 pixel_rate;
 	u32 line_rate;
 
-	/* Currently only 6 tabs supported */
-	chan_ctx->xm2msc_dev->taps = XSCALER_TAPS_6;
 	xm2mvsc_initialize_coeff_banks(chan_ctx);
 
 	pixel_rate = (out_q_data->width * XM2MSC_STEP_PRECISION) /
@@ -1832,6 +1830,16 @@ static int xm2msc_parse_of(struct platform_device *pdev,
 		return -EINVAL;
 	}
 
+	ret = of_property_read_u32(node, "xlnx,num-taps",
+				   &xm2msc->taps);
+	if (ret || (xm2msc->taps != XSCALER_TAPS_6 &&
+		    xm2msc->taps != XSCALER_TAPS_8 &&
+		    xm2msc->taps != XSCALER_TAPS_10 &&
+		    xm2msc->taps != XSCALER_TAPS_12)) {
+		dev_err(dev, "missing/invalid taps in dts prop\n");
+		return -EINVAL;
+	}
+
 	xm2msc->irq = irq_of_parse_and_map(node, 0);
 	if (xm2msc->irq < 0) {
 		dev_err(dev, "Unable to get IRQ");
@@ -1842,6 +1850,7 @@ static int xm2msc_parse_of(struct platform_device *pdev,
 	dev_dbg(dev, "DMA Addr width Supported = %d\n", xm2msc->dma_addr_size);
 	dev_dbg(dev, "Max col/row Supported = (%d) / (%d)\n",
 		xm2msc->max_wd, xm2msc->max_ht);
+	dev_dbg(dev, "taps Supported = %d\n", xm2msc->taps);
 	/* read supported video formats and update internal table */
 	hw_vid_fmt_cnt = of_property_count_strings(node, "xlnx,vid-formats");
 
-- 
2.31.1

